shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java

```diff
@@ -1310,7 +1311,7 @@ public void inputConnection_textInputTypeNone() {
     InputConnection connection =
         textInputPlugin.createInputConnection(
             testView, mock(KeyboardManager.class), new EditorInfo());
-    assertEquals(connection, null);
+    assertNotNull(connection);
```

Comment:
I don't think this pr has the correct level of testing for this bug.  I would expect either a new test that simulates some version of input and verifies it gets sent when the type is NONE or an existing test being modified from ensuring text was not sent to text being sent.

Reply:
Agreed. I'll try to find some time over the next few days to find a test case I can piggy back off of.

Reply:
+1 to needing a proper test case that fails without your change to TextInputPlugin.java applied.

Reply:
@johnmccutchan @reidbaker added additional test case. I admittedly just copied the existing `inputConnection_createsActionFromEnter` test case and changed the `TextInputType` to `NONE` which caused a NullPointerException within the test case itself (source of the bug, no connection created). Obviously after updates, test case passes.

Reply:
Could you refactor the original test into a function that accepts TextInputType as an argument and avoid the code duplication?

Reply:
Thanks for making the change!